compileall2.py

```diff
@@ -256,14 +264,22 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
             if not quiet:
                 print('Compiling {!r}...'.format(fullname))
             try:
-                for opt_level, cfile in opt_cfiles.items():
+                for index, opt_level in enumerate(sorted(optimize)):
+                    cfile = opt_cfiles[opt_level]
                     if PY37:
                         ok = py_compile.compile(fullname, cfile, dfile, True,
                                                 optimize=opt_level,
                                                 invalidation_mode=invalidation_mode)
                     else:
                         ok = py_compile.compile(fullname, cfile, dfile, True,
                                                 optimize=opt_level)
+                    
+                    if index > 0 and hardlink_dupes:
+                        previous_cfile = opt_cfiles[optimize[index - 1]]
+                        if  filecmp.cmp(cfile, previous_cfile):
```

Comment:
```suggestion
                        if  filecmp.cmp(cfile, previous_cfile, shallow=False):
```

This is needed to check the content and not just `os.stat()` signatures.